darts/tests/models/forecasting/test_torch_forecasting_model.py

```diff
@@ -216,6 +219,215 @@ def test_manual_save_and_load(self):
                 model_chained_load_save.predict(n=4), model_manual_save.predict(n=4)
             )
 
+        def test_save_and_load_weights_w_encoders(self):
+            """
+            Verify that save/load does not break encoders.
+
+            Note: since load_weights() calls load_weights_from_checkpoint(), it will be used
+            for all but one test.
+            Note: Using DLinear since it supports both past and future covariates
+            """
+
+            def create_DLinearModel(
+                model_name: str,
+                save_checkpoints: bool = False,
+                add_encoders: Dict = None,
+            ):
+                return DLinearModel(
+                    input_chunk_length=4,
+                    output_chunk_length=1,
+                    kernel_size=5,
+                    model_name=model_name,
+                    add_encoders=add_encoders,
+                    work_dir=self.temp_work_dir,
+                    save_checkpoints=save_checkpoints,
+                    random_state=42,
+                    force_reset=True,
+                )
+
+            model_dir = os.path.join(self.temp_work_dir)
+            manual_name = "save_manual"
+            auto_name = "save_auto"
+            auto_name_other = "save_auto_other"
+            # create manually saved model checkpoints folder
+            checkpoint_path_manual = os.path.join(model_dir, manual_name)
+            os.mkdir(checkpoint_path_manual)
+            checkpoint_file_name = "checkpoint_0.pth.tar"
+            model_path_manual = os.path.join(
+                checkpoint_path_manual, checkpoint_file_name
+            )
+
+            # define encoders sets
+            encoders_past = {
+                "datetime_attribute": {"past": ["day"]},
+                "transformer": Scaler(),
```

Comment:
can the user also load weights with `load_encoders=False` and the encoders use a different transformer than used when fitting the model?